codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,482 @@
+                return false;
+            }
+        }
+
+        if (localYear > 0) {
+            year = localYear;
+            return true;
+        }
+
+        return false;
+    }
+
+    private void parseToken(CharSequence txt, int tokenStart, int tokenEnd) {
+        if (!timeFound) {
+            timeFound = tryParseTime(txt, tokenStart, tokenEnd);
+            if (timeFound) {
+                return;
+            }
+        }
+
+        if (!dayOfMonthFound) {
+            dayOfMonthFound = tryParseDayOfMonth(txt, tokenStart, tokenEnd);
+            if (dayOfMonthFound) {
+                return;
+            }
+        }
+
+        if (!monthFound) {
+            monthFound = tryParseMonth(txt, tokenStart, tokenEnd);
+            if (monthFound) {
+                return;
+            }
+        }
+
+        if (!yearFound) {
+            yearFound = tryParseYear(txt, tokenStart, tokenEnd);
+        }
+    }
+
+    private Date parse0(CharSequence txt, int start, int end) {
+        int tokenStart = -1;
+
+        for (int i = start; i < end; i++) {
+            char c = txt.charAt(i);
+
+            if (isDelim(c)) {
+                if (tokenStart != -1) {
+                    // terminate token
+                    parseToken(txt, tokenStart, i);
+                    tokenStart = -1;
+                }
+            } else {
+                if (tokenStart == -1) {
+                    // start new token
+                    tokenStart = i;
+                }
+            }
+        }
+
+        if (tokenStart != -1) {
+            // terminate trailing token
+            parseToken(txt, tokenStart, txt.length());
+        }
+
+        if (!timeFound || !dayOfMonthFound || !monthFound || !yearFound
+                || dayOfMonth < 1 || dayOfMonth > 31 || hours > 23 || minutes > 59 || seconds > 59) {
+            // missing field or invalid values
+            return null;
+        }
+
+        if (year >= 70 && year <= 99) {
+            year += 1900;
+        } else if (year >= 0 && year < 70) {
+            year += 2000;
+        } else if (year < 1601) {
+            // invalid value
+            return null;
+        }
+
+        cal.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+        cal.set(Calendar.MONTH, month);
+        cal.set(Calendar.YEAR, year);
+        cal.set(Calendar.HOUR_OF_DAY, hours);
+        cal.set(Calendar.MINUTE, minutes);
+        cal.set(Calendar.SECOND, seconds);
+        return cal.getTime();
+    }
+
+    private String format0(Date date) {
+        append0(date, sb);
+        return sb.toString();
+    }
+
+    private StringBuilder append0(Date date, StringBuilder sb) {
+        cal.setTime(date);
+
+        switch (cal.get(Calendar.DAY_OF_WEEK)) {
+            case Calendar.MONDAY:
+                sb.append("Mon, ");
+                break;
+            case Calendar.TUESDAY:
+                sb.append("Tue, ");
+                break;
+            case Calendar.WEDNESDAY:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)